include/log.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <string>
+#include <misc_log_ex.h>
+
+struct mg_mgr;
+struct mg_connection;
+
+namespace graft
+{
+    extern std::string client_addr(mg_connection* client);
+} //namespace graft
+
+#define LOG_PRINT_CLN(level,client,x) LOG_PRINT_L##level("[" << client_addr(client) << "]" << x)
```

Comment:
If those are ClientTask/BaseTask -specific, it makes better sense to move them to task.h